packages/ee/lib/service.ts

```diff
@@ -89,47 +89,86 @@ const fetchLicenseForE2ETesting = async (): Promise<{
   }
 };
 
-export const getIsEnterpriseEdition = async (): Promise<boolean> => {
+export const getEnterpriseLicense = async (): Promise<{
+  active: boolean;
+  features: TEnterpriseLicenseFeatures | null;
+  lastChecked: Date;
+  isPendingDowngrade?: boolean;
+}> => {
   if (!ENTERPRISE_LICENSE_KEY || ENTERPRISE_LICENSE_KEY.length === 0) {
-    return false;
+    return {
+      active: false,
+      features: null,
+      lastChecked: new Date(),
+    };
   }
 
   if (E2E_TESTING) {
     const previousResult = await fetchLicenseForE2ETesting();
-    return previousResult && previousResult.active !== null ? previousResult.active : false;
+    // return previousResult && previousResult.active !== null ? previousResult.active : false;
+    return {
+      active: previousResult ? previousResult.active : false,
+      features: previousResult ? previousResult.features : null,
+      lastChecked: previousResult ? previousResult.lastChecked : new Date(),
+    };
   }
 
   // if the server responds with a boolean, we return it
   // if the server errors, we return null
   // null signifies an error
   const license = await fetchLicense();
+
   const isValid = license ? license.status === "active" : null;
+  const threeDaysInMillis = 3 * 24 * 60 * 60 * 1000;
+  const currentTime = new Date();
 
   const previousResult = await getPreviousResult();
+
   if (previousResult.active === null) {
     if (isValid === null) {
-      await setPreviousResult({
+      const newResult = {
         active: false,
         features: { isMultiOrgEnabled: false },
         lastChecked: new Date(),
-      });
-      return false;
+      };
+
```

Comment:
I think we need to return `isPendingDowngrade: true,` from here as well, in case the license check fails more than once.